### PR TITLE
Add FUTEX_WAIT_BITSET and FUTEX_WAKE_BITSET

### PR DESCRIPTION
**Pull request by liqinggd**
Depends on https://github.com/occlum/linux-sgx/pull/8, remember to merge the PR in SGX SDK first

### CODE REVIEW

File: src/libos/src/process/syscalls.rs [hunk 1]
```diff
@@ -141,27 +142,39 @@ pub fn do_futex(
...
 
+    let get_futex_timeout = |timeout| -> Result<Option<FutexTimeout>> {
+        let timeout = timeout as *const timespec_t;
+        if timeout.is_null() {
+            return Ok(None);
+        }
+        let ts = timespec_t::from_raw_ptr(timeout)?;
+        ts.validate()?;
+        let clock_id = if futex_flags.contains(FutexFlags::FUTEX_CLOCK_REALTIME) {
+            ClockID::CLOCK_REALTIME
```
Comment (guzongmin) [inline src/libos/src/process/syscalls.rs:153]: Please add a To-do task here. We need to evaluate the security impact when using CLOCK_REALTIME.

Comment (liqinggd) [reply src/libos/src/process/syscalls.rs:154]: Done

**Review by guzongmin — DISMISSED** [review]
LGTM

**Review by tatetian — DISMISSED** [review]
LGTM. Thanks for the contribution!

**Comment by liqinggd** [comment]
Find a bug in the previous commit, already updated in this commit, please review.

The bug is as follows:
the `timout_ptr` is a temporary value in the closure, see the original commit below:
image

**Review by guzongmin — APPROVED** [review]

**Pull request merged**